lib/icinga/icingaapplication.cpp

```diff
@@ -138,18 +140,22 @@ static void PersistModAttrHelper(std::fstream& fp, ConfigObject::Ptr& previousOb
 		});
 		ConfigWriter::EmitFunctionCall(fp, "get_object", args1);
 
-		ConfigWriter::EmitRaw(fp, "\nif (obj) {\n");
+		ConfigWriter::EmitRaw(fp, "\nif (obj) {\n\tvar result\n");
 	}
 
-	ConfigWriter::EmitRaw(fp, "\tobj.");
+	ConfigWriter::EmitRaw(fp, "\tif (obj.get_attribute(");
+	ConfigWriter::EmitString(fp, attr);
+	ConfigWriter::EmitRaw(fp, ", &result) && Json.encode(Internal.serialize(result, 2 /* FAConfig */)) == ");
```

Comment:
Given that `obj.get_attribute(...)` will return false if the attribute does not exist in the original object (i.e. as read from the regular config files), this will lose modified attributes in this case.

Also, is JSON-encoding things really the best we can do to compare values?

Reply:
1. Let’s say: "if the attribute does not exist in the original object **anymore**" – that’s this PR's intention
2. Inside config files IMAO yes